ODSP driver: TreesLatest telemetry cleanup

NicholasCouri opened the pull request:
Fixes #9181 
From the bug: 
1. We should remove overalltime property. We have duration that tracks same thing.
**(nichoc)** removed overalltime property
2. fetchstarttorespendtime, and others - should it be "respond", not "respend"? Also should it be fetchStartToRespondTime (camal case to make it more readable)?
**(nichoc)** Renaming - fecthStartToResponseEndTime
3. At 95% percentile, I see responsenetworkTime > duration. This seems really odd. I'd expect for any session that to be not the case, and thus true at any percentile. It's possible that some events do not have one or the other property and that's what makes it work that way. We need to get to the bottom of it.
**(nichoc)** There are scenarios where requestStart and responseStart can return 0 -  #do not calculate the reqStartToResponseEndTime and responseNetworkTime in these cases. 
4. fetchstarttorespendtime - I find name not telling me what it is. Looking at the code, it sounds like it's pretty much the same as networktime? What's the difference? Can we someone capture it in the name?
**(nichoc)** - Added comments in the code to allow understanding of these fields.
5. This code (and similar code) should not log zeros, it should log undefined if we do not have data:
fetchStToRespEndTime = (indResTime.fetchStart > 0) ?
(indResTime.responseEnd - indResTime.fetchStart) : 0;
**(nichoc)** changed to undefined if we do not have data. 
@andre4i @vladsud FYI

NicholasCouri committed:
Fix for bug 9181 - ODSP driver: TreesLatest telemetry cleanup

vladsud commented on packages/drivers/odsp-driver/src/fetchSnapshot.ts:
```diff
@@ -303,16 +310,14 @@ async function fetchLatestSnapshotCore(
-                    overalltime: overallTime,
-                    networktime: networkTime,
-                    clienttime: clientTime,
+                    redirectTime: redirectTime,
```
I think compiler should fail and ask to use more concise form:
```ts
redirectTime,
```
It might be worth taking a look at other variable names and rename them to be what property in telemetry should look like to make code more compact.

NicholasCouri replied:
renamed them.

vladsud commented on packages/drivers/odsp-driver/src/fetchSnapshot.ts:
```diff
@@ -249,14 +255,16 @@ async function fetchLatestSnapshotCore(
+                            (indResTime.responseEnd - indResTime.requestStart) : undefined;
+                        networkTime = (indResTime.startTime > 0) ?
+                            (indResTime.responseEnd - indResTime.fetchStart) : undefined;
+                        if (spReqDuration && networkTime) {
```
it's better for checks to be explicit, i.e. spReqDuration  !== undefined.
The way it's written it also will be falsey if the value is zero - I think it's not the intention here

vladsud commented on packages/drivers/odsp-driver/src/fetchSnapshot.ts:
```diff
@@ -303,16 +310,14 @@ async function fetchLatestSnapshotCore(
-                    overalltime: overallTime,
-                    networktime: networkTime,
-                    clienttime: clientTime,
+                    redirectTime: redirectTime,
```
Like you comment above with description of what browser provides, but it would be great to add similar comment that says what each telemetry property means. Such that developers looking at telemetry can quickly reason about it, without need to parse code above and understand transformation we do.

NicholasCouri committed:
Addressing review comments:
- Rename variables to macth the telemetry fields.
- Use if (var !== undefined) instead of if (var).
- Add description of the telemetry fields.
